src/python/CMSMonitoring/StompAMQ.py

```diff
@@ -270,6 +292,6 @@ def make_notification(self, payload, docType, docId=None, producer=None, ts=None
         notification.update(headers)
         notification['body'] = body
 
-        if returnValidationResult:
-            return notification, validated
+        if returnOffendingKeys:
+            return notification, offending_keys, unknown_keys
```

Comment:
Here you return two lists in addition to notification, while on line below it is only notification. Usually different return metrics may cause more trouble upstream since clients need to know how may return parameters to expect and their code may break depending on run-time conditions. For example, let's say client call is
```
    returnOffendingKeys = config.get() # client get its parameter at run time
    notification = make_notification(..., returnOffendingKeys)
```
this code will not behave as expecting if `returnOffendingKeys` parameter change at run-time. Since when it is not set code will return one notification parameter, while when it is set it will get three parameters.

It is much safer to have consistent set of return parameters which consistent data-types.

Reply:
Agreed, this isn't very pretty. We introduced the `returnValidationResult` to be backwards compatible, but I'm happy to remove it and just return the lists in any case.